packages/runtime/datastore/src/localChannelContext.ts

```diff
@@ -50,6 +50,9 @@ export abstract class LocalChannelContextBase implements IChannelContext {
                 readonly objectStorage: ChannelStorageService;
             }>,
     ) {
+        if (this.id.includes("/")) {
+            throw new UsageError("Ids for LocalChannelContextBase cannot have '/' characters");
```

Comment:
Should we maybe make this just an `assert` at the channel level then, and throw the `UsageError` from the higher-level abstractions? `create()`  methods on DDS types or the ones on data store that would end up calling this. So the DDS concept doesn't leak down to the "plumbing".

Reply:
That was my initial thought too but the problem with that is that you have to do for every DDS. And if new DDS are added, they can miss it. So, we want this to be a central place that every DDS creation goes through.
SharedObject may be closer to the DDS where this can be added. Another perspective may be that having in channel context makes it consistent with the same check in data store layer.